x-pack/plugins/viz_editor/bar_chart_plugin/bar_chart_plugin.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { columnSummary } from '../public/common/components/config_panel';
+import { IndexPatternPanel } from '../public/common/components/index_pattern_panel';
+import { Axis, selectColumn, ViewModel } from '../public/common/lib';
+import { EditorPlugin, PanelComponentProps } from '../public/editor_plugin_registry';
+
+interface BarChartVisState extends ViewModel {
```

Comment:
This is an important part: The "extensions" of the view model are defined in the individual plugins - the basic view model from the editor frame only contains the lowest common denominator.